src/common_auth.cpp

```diff
@@ -61,6 +61,61 @@ char* s3fs_base64(unsigned char* input, size_t length)
   return result;
 }
 
+inline unsigned char char_decode64(const char ch)
+{
+  unsigned char by;
+  if('A' <= ch && ch <= 'Z'){                   // A - Z
+    by = static_cast<unsigned char>(ch - 'A');
+  }else if('a' <= ch && ch <= 'z'){             // a - z
+    by = static_cast<unsigned char>(ch - 'a' + 26);
+  }else if('0' <= ch && ch <= '9'){             // 0 - 9
+    by = static_cast<unsigned char>(ch - '0' + 52);
+  }else if('+' == ch){                         // +
+    by = 62;
+  }else if('/' == ch){                         // /
+    by = 63;
+  }else if('=' == ch){                         // =
+    by = 64;
+  }else{                                       // something wrong
+    by = 64;
```

Comment:
I think that we should not terminate s3fs as a character has been detected illegal.
Because it is base64 string that has been written in the http header, there is a possibility that illegal character exists.
It will skip the character instead, and it will the act is another error(ex. not found extended attribute)

Thanks.